lib/ClangImporter/ImportDecl.cpp

```diff
@@ -3392,14 +3396,19 @@ namespace {
       }
 
       HeaderLoc loc(decl->getLocation());
-      if (isForeignReferenceTypeWithoutImmortalAttrs(decl->getReturnType())) {
+      const auto retType =
+          isa<clang::FunctionDecl>(decl)
+              ? dyn_cast<clang::FunctionDecl>(decl)->getReturnType()
```

Comment:
Let's use `cast` instead of `dyn_cast` here as we are confident about the type of the node.